### PR TITLE
Dashing devel

### PR DESCRIPTION
**Pull request by brightenlee**
Ported except joystick_relay node

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,57 +1,87 @@
...
-   DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}
-   FILES_MATCHING PATTERN "*.h"
+install(
+  DIRECTORY include/
+  DESTINATION include
 )
 
-if (CATKIN_ENABLE_TESTING)
-    find_package(rostest REQUIRED)
+if (BUILD_TESTING)
```
Comment (bmagyar) [inline CMakeLists.txt:65]: please either uncomment or delete the commented stuff

File: include/twist_mux/twist_mux.hpp [hunk 2]
```diff
@@ -41,60 +46,58 @@ class LockTopicHandle;
  * @brief The TwistMux class implements a top-level twist multiplexer module
  * that priorize different velocity command topic inputs according to locks.
  */
-class TwistMux
+class TwistMux : public rclcpp::Node
 {
 public:
+  // RCLCPP_SMART_PTR_DEFINITIONS(TwistMux);
```
Comment (bmagyar) [inline include/twist_mux/twist_mux.hpp]: commented stuff

**Review by bmagyar — CHANGES_REQUESTED** [review]
Unfortunately github is not intelligent enough to connect the files you renamed in the diff view. Please undo the renaming of `.h` files to `.hpp` so I can properly assess the changes.

File: test/test_foo.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+def test_foo():
```
Comment (bmagyar) [inline test/test_foo.py]: how about removing this?

File: test/system_blackbox.py [hunk 4]
```diff
@@ -64,26 +66,31 @@ def setUpClass(cls):
...
+        #
+        # TODO(artivis) use rate once available
+        time.sleep(self.MESSAGE_TIMEOUT + self.TOPIC_TIMEOUT)
 
     @classmethod
     def _vel_cmd(cls):
-        rospy.sleep(cls.MESSAGE_TIMEOUT)
+        # TODO(artivis) use rate once available
+        time.sleep(cls.MESSAGE_TIMEOUT)
+        # TODO wait_for_msg-like functionnality not yet available
```
Comment (bmagyar) [inline test/system_blackbox.py:92]:
```suggestion
        # TODO wait_for_msg-like functionality not yet available
```

File: test/system_blackbox.py [hunk 5]
```diff
@@ -64,26 +66,31 @@ def setUpClass(cls):
...
+        # https://github.com/ros2/rclcpp/issues/520
```
Comment (bmagyar) [inline test/system_blackbox.py:93]: These in-code TODO-s will be forgotten and never addressed. Could you please instead open an issue on github with some detail and put `TODO: link....` here?

File: test/system_blackbox.py [hunk 6]
```diff
@@ -64,26 +66,31 @@ def setUpClass(cls):
...
+        self._lock2.pub(unlock)
 
         # Wait for previously published messages to time out,
         # since we aren't restarting twist_mux.
         #
         # This sleeping time must be higher than any of the
         # timeouts in system_test_config.yaml.
-        rospy.sleep(self.MESSAGE_TIMEOUT + self.TOPIC_TIMEOUT)
+        #
+        # TODO(artivis) use rate once available
```
Comment (bmagyar) [inline test/system_blackbox.py:85]: These in-code TODO-s will be forgotten and never addressed. Could you please instead open an issue on github with some detail and put `TODO: link....` here?

File: src/twist_mux_diagnostics.cpp [hunk 7]
```diff
@@ -16,80 +16,74 @@
...
-  stat.add("current priority", static_cast<int>(status_.priority));
+  stat.add("current priority", static_cast<int>(status_->priority));
 
-  stat.add("loop time in [sec]", status_.main_loop_time);
-  stat.add("data age in [sec]", status_.reading_age);
+  stat.add("loop time in [sec]", status_->main_loop_time);
+  stat.add("data age in [sec]", status_->reading_age);
 
-  ROS_DEBUG_THROTTLE(1.0, "Publishing diagnostics.");
+  //  ROS_DEBUG_THROTTLE(1.0, "Publishing diagnostics.");
```
Comment (bmagyar) [inline src/twist_mux_diagnostics.cpp]: commented code, please eithe rremove or fix

**Comment by M-Redel** [comment]
how about with ROS2 eloquent?

**Comment by Michael-Equi** [comment]
@bmagyar  Is this ready to be merged yet? Looks like it has been sitting a bit and the ros-planning/dashing-devel branch is still at ROS1 stuff.

**Pull request closed without merging**